src/main/java/teammates/ui/webapi/action/UpdateStudentAction.java

```diff
@@ -0,0 +1,99 @@
+import teammates.common.exception.TeammatesException;
+import teammates.common.exception.UnauthorizedAccessException;
+import teammates.common.util.Const;
+import teammates.common.util.EmailWrapper;
+import teammates.common.util.Logger;
+import teammates.logic.api.EmailGenerator;
+import teammates.ui.webapi.request.StudentUpdateRequest;
+
+/**
+ * Action: Edits details of a student in a course.
+ */
+public class UpdateStudentAction extends Action {
+    private static final Logger log = Logger.getLogger();
+
+    @Override
+    protected AuthType getMinAuthLevel() {
+        return authType.LOGGED_IN;
+    }
+
+    @Override
+    public void checkSpecificAccessControl() {
+        if (!userInfo.isInstructor) {
+            throw new UnauthorizedAccessException("Instructor privilege is required to access this resource.");
+        }
+        String courseId = getRequestParamValue(Const.ParamsNames.COURSE_ID);
+
+        InstructorAttributes instructor = logic.getInstructorForGoogleId(courseId, userInfo.id);
+        gateKeeper.verifyAccessible(
+                instructor, logic.getCourse(courseId), Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_STUDENT);
+    }
+
+    @Override
+    public ActionResult execute() {
+        String courseId = getNonNullRequestParamValue(Const.ParamsNames.COURSE_ID);
+        String studentEmail = getNonNullRequestParamValue(Const.ParamsNames.STUDENT_EMAIL);
+
+        StudentAttributes student = logic.getStudentForEmail(courseId, studentEmail);
+        if (student == null) {
+            return new JsonResult(Const.StatusMessages.STUDENT_NOT_FOUND_FOR_EDIT, HttpStatus.SC_NOT_FOUND);
+        }
+
+        StudentUpdateRequest updateRequest = getAndValidateRequestBody(StudentUpdateRequest.class);
+        try {
+            logic.validateSectionsAndTeams(Arrays.asList(student), student.course);
+            logic.validateSectionsAndTeams(Arrays.asList(student), student.course);
+            logic.updateStudentCascade(
+                    StudentAttributes.updateOptionsBuilder(courseId, studentEmail)
+                            .withName(updateRequest.getName())
+                            .withNewEmail(updateRequest.getEmail())
+                            .withTeamName(updateRequest.getTeam())
+                            .withSectionName(updateRequest.getSection())
+                            .withComment(updateRequest.getComments())
+                            .build());
+
+            StringBuilder responseMessage = new StringBuilder("Student has been updated");
+            if (!student.email.equals(updateRequest.getEmail())) {
+                logic.resetStudentGoogleId(updateRequest.getEmail(), courseId);
+                boolean isSessionSummarySendEmail =
+                        getBooleanRequestParamValue(Const.ParamsNames.SESSION_SUMMARY_EMAIL_SEND_CHECK);
+
+                if (isSessionSummarySendEmail && sendEmail(courseId, updateRequest.getEmail())) {
```

Comment:
I am not a big fan of using a boolean method like `sendEmail` to execute an action. Boolean methods are generally assumed to not have any side effects. You can read this for more detail: https://www.teamten.com/lawrence/programming/keep-if-clauses-side-effect-free.html

Please re-inline `sendEmail` into the `execute` method. You probably need to rearrange the message appending around after doing this.

Reply:
Yeah your right, it isnt the best practice. I will update accordingly!